monitoring/persistent_stats_history.cc

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#include "monitoring/persistent_stats_history.h"
+
+#include <cstring>
+#include <string>
+#include <utility>
+#include "db/db_impl/db_impl.h"
+#include "port/likely.h"
+#include "util/string_util.h"
+
+namespace rocksdb {
+// 10 digit seconds timestamp => [Sep 9, 2001 ~ Nov 20, 2286]
+const int kNowSecondsStringLength = 10;
+const int kMicrosInSecond = 1000 * 1000;
+const char* kFormatVersionKeyString = "__persistent_stats_format_version__";
+const char* kReaderVersionKeyString = "__persistent_stats_reader_version__";
+const int kStatsCFCurrentFormatVersion = 1;
+const int kStatsCFMinimumFormatVersion = 1;
+const int kPersistentStatsReaderVersion = 1;
+
+void GetVersionKeyString(char* buf, StatsVersionKeyType type) {
+  switch (type) {
+    case StatsVersionKeyType::kFormatVersion:
+      strcpy(buf, kFormatVersionKeyString);
+      return;
+    case StatsVersionKeyType::kReaderVersion:
+      strcpy(buf, kReaderVersionKeyString);
+      return;
+    default:
+      return;
+  }
+}
+
+size_t EncodePersistentStatsVersionKey(char* buf, StatsVersionKeyType type) {
+  if (type >= StatsVersionKeyType::kKeyTypeMax) {
+    return -1;
+  }
+  GetVersionKeyString(buf, type);
+  return strlen(buf);
+}
+
+int DecodePersistentStatsVersionNumber(DBImpl* db,
+                                       StatsVersionKeyType type) {
+  if (type >= StatsVersionKeyType::kKeyTypeMax) {
+    return -1;
+  }
+  char key_string[100];
+  GetVersionKeyString(key_string, type);
```

Comment:
Similarly, it's too complicated. If we make `kFormatVersionKeyString` an std::string, maybe we just can do `Get(options, kFormatVersionKeyString)`.